.github/workflows/awesome_workflow.yml

```
@@ -40,12 +40,16 @@ jobs:
       - name: Lint modified files
         shell: bash
         run: python3 scripts/file_linter.py
-      - name: Commit and push changes
+      - if: github.repository == 'TheAlgorithms/C-Plus-Plus' && github.ref == 'refs/heads/master'
+        name: Commit, push, and create a PR
         run: |
-          git diff DIRECTORY.md
-          git commit -am "clang-format and clang-tidy fixes for ${GITHUB_SHA::8}" || true
-          git push origin HEAD:$GITHUB_REF || true
+          git checkout -b clang-tidy-${{ github.sha }}
+          git commit -am "fix: clang-format and clang-tidy fixes for ${GITHUB_SHA::8}"
```

maybe we use lint: instead of fix:
```suggestion
          git commit -am "lint: clang-format and clang-tidy fixes for ${GITHUB_SHA::8}"

although [conventional commits](```
https://www.conventionalcommits.org/en/v1.0.0-beta.4/#specification) dont recognize it. I've seen alots of repos use either lint:

---

We should stick to `fix` for now. We can always change it later on. 🙂

---

seems good to me!